test/org/sagebionetworks/bridge/hibernate/HibernateAccountDaoTest.java

```diff
@@ -2147,6 +2158,220 @@ public void authenticateAccountEmailUnverifiedWithoutEmailVerificationOK() throw
+    // exception and convert it to the Bridge namespace.
+    @Test(expected = org.sagebionetworks.bridge.exceptions.ConstraintViolationException.class)
+    public void genericConstraintViolationOnCreate() {
+        PersistenceException ex = new PersistenceException(mock(ConstraintViolationException.class));
+        doThrow(ex).when(mockHibernateHelper).create(any());
+        
+        GenericAccount account = makeValidGenericAccount();
+        dao.createAccount(study, account);
+    }
+    
+    @Test(expected = org.sagebionetworks.bridge.exceptions.ConstraintViolationException.class)
+    public void genericConstraintViolationOnUpdate() throws Exception {
+        PersistenceException ex = new PersistenceException(mock(ConstraintViolationException.class));
+        doThrow(ex).when(mockHibernateHelper).update(any());
+        
+        HibernateAccount hibernateAccount = makeValidHibernateAccount(false, false);
+        when(mockHibernateHelper.getById(eq(HibernateAccount.class), any())).thenReturn(hibernateAccount);
+         GenericAccount account = makeValidGenericAccount();
+        dao.updateAccount(account);
+    }
+    
+    @Test
+    public void emailConstraintViolationOnCreate() throws Exception {
+        ConstraintViolationException cve = new ConstraintViolationException(
+                "Duplicate entry ABC for key 'Accounts-StudyId-Email-Index'", null, null);
+        PersistenceException ex = new PersistenceException(cve);
+        doThrow(ex).when(mockHibernateHelper).create(any());
+     
+        GenericAccount account = makeValidGenericAccount();
+        account.setId("id");
+        
+        HibernateAccount hibernateAccount = makeValidHibernateAccount(false, false);
+        hibernateAccount.setId("id2");
+        when(mockHibernateHelper.queryGet(any(), any(), any(), any(), eq(HibernateAccount.class)))
+                .thenReturn(ImmutableList.of(hibernateAccount));
+        try {
+            dao.createAccount(study, account);
+            fail("Should have thrown an exception");
+        } catch(EntityAlreadyExistsException e) {
+            assertEquals("Email address has already been used by another account.", e.getMessage());
+            assertEquals("id2", e.getEntityKeys().get("userId"));
+        }
+    }
+    
+    @Test
+    public void emailConstraintViolationOnUpdate() throws Exception {
+        ConstraintViolationException cve = new ConstraintViolationException(
+                "Duplicate entry ABC for key 'Accounts-StudyId-Email-Index'", null, null);
+        PersistenceException ex = new PersistenceException(cve);
+        doThrow(ex).when(mockHibernateHelper).update(any());
+        
+        GenericAccount account = makeValidGenericAccount();
+        account.setId("id");
+        
+        HibernateAccount hibernateAccount = makeValidHibernateAccount(false, false);
+        hibernateAccount.setId("id2");
+        when(mockHibernateHelper.getById(any(), any())).thenReturn(hibernateAccount);
```

Comment:
This test returns account2 when we're trying to update account1. This would never happen in prod (I hope), but it results in a nonsensical test. It's also probably why the incorrect IDs issue wasn't caught by tests.

This as well as phoneConstraintViolationOnUpdate() and externalIdConstraintViolationOnUpdate().